From 80389fb1f8f0e0be533ece56db42e9caaebd9393 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Fri, 30 Jul 2010 19:29:52 +0800
Subject: [PATCH] ENGR00125771 usb:should open usb related clock at usb wakeup enable mode

Fix the bug that not open usb related clock at usb wakeup enable mode

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 drivers/usb/gadget/arcotg_udc.c |   13 ++++++-------
 1 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/usb/gadget/arcotg_udc.c b/drivers/usb/gadget/arcotg_udc.c
index f416902..9947da5 100644
--- a/drivers/usb/gadget/arcotg_udc.c
+++ b/drivers/usb/gadget/arcotg_udc.c
@@ -2205,14 +2205,13 @@ int usb_gadget_register_driver(struct usb_gadget_driver *driver)
 	udc_controller->gadget.dev.driver = &driver->driver;
 	spin_unlock_irqrestore(&udc_controller->lock, flags);
 
-	if (!device_may_wakeup(udc_controller->gadget.dev.parent)) {
-		if (udc_controller->pdata->usb_clock_for_pm)
-			udc_controller->pdata->usb_clock_for_pm(true);
+	if (udc_controller->pdata->usb_clock_for_pm)
+		udc_controller->pdata->usb_clock_for_pm(true);
+
+	portsc = fsl_readl(&dr_regs->portsc1);
+	portsc &= ~PORTSCX_PHY_LOW_POWER_SPD;
+	fsl_writel(portsc, &dr_regs->portsc1);
 
-		portsc = fsl_readl(&dr_regs->portsc1);
-		portsc &= ~PORTSCX_PHY_LOW_POWER_SPD;
-		fsl_writel(portsc, &dr_regs->portsc1);
-	}
 	/* bind udc driver to gadget driver */
 	retval = driver->bind(&udc_controller->gadget);
 	if (retval) {
-- 
1.5.4.4

